fictadvisor-api/src/v2/utils/files/FileService.ts

```diff
@@ -1,51 +1,67 @@
 import { Injectable } from '@nestjs/common';
-import { Express } from 'express';
 import { createHash } from 'crypto';
 import { join, extname, resolve as pathResolve } from 'path';
-import { resolve } from 'url';
 import PizZip from 'pizzip';
 import Docxtemplater from 'docxtemplater';
-import * as fs from 'fs';
+import * as admin from 'firebase-admin';
 import * as process from 'process';
-import { v4 } from 'uuid';
 import { find } from '../ArrayUtil';
 import { StudentWithContactsData } from '../../api/datas/StudentWithContactsData';
 import { MINUTE } from '../date/DateService';
+import { Bucket } from '@google-cloud/storage';
 
 @Injectable()
 export class FileService {
-  async saveByHash (file: Express.Multer.File, directory: string) {
-    const fileName = createHash('md5').update(file.buffer).digest('hex');
-    const filePath = join(pathResolve(), 'static', directory, fileName + extname(file.originalname));
+  private storage: Bucket;
 
-    await fs.promises.writeFile(filePath, file.buffer);
+  constructor () {
+    admin.initializeApp({
+      credential: admin.credential.cert(process.env.GOOGLE_APPLICATION_CREDENTIALS),
+      storageBucket: process.env.BUCKET_NAME,
+    });
+    this.storage = admin.storage().bucket();
+  }
+
+  async saveByHash (fileContent: Express.Multer.File, directory: string) {
+    const fileName = createHash('md5').update(fileContent.buffer).digest('hex');
+    const filePath = join('static', directory, fileName + extname(fileContent.originalname));
 
-    return resolve(process.env.BASE_URL, join(directory, fileName + extname(file.originalname)));
+    const file = this.storage.file(filePath);
+    await file.save(fileContent.buffer);
+
+    const [url] = (await file.getSignedUrl({
+      action: 'read',
+      expires: '01-01-2222',
+    }));
+    return url;
   }
 
   getPathFromLink (link: string): string {
     const url = new URL(link);
     return url.pathname;
   }
 
-  checkFileExist (path: string, isPrivate = true): boolean {
+  async checkFileExist (path: string, isPrivate = true): Promise<boolean> {
     const filePath = join(pathResolve(), isPrivate ? 'private' : 'static', path);
```

Comment:
We now save files without the path to the working directory, don't we?
```suggestion
    const filePath = join(isPrivate ? 'private' : 'static', path);
```